### DISCUSSION TITLE
I am getting a persistent error message in my terminal: no such file or directory: /opt/homebrew/bin/brew

### DISCUSSION THREAD

**Discussion by michellehutton** [Homebrew/discussions#4578] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.23
ORIGIN: https://github.com/Homebrew/brew
HEAD: 2c95aa49563dbf322672423cbfcf3d17ea434d2c
Last commit: 33 hours ago
Core tap JSON: 17 Jun 19:23 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit cometlake
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4-x86_64
CLT: 14.3.1.0.1.1683849156
Xcode: 14.3.1
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I am getting a persistent error message in my terminal (but only on my 2020 iMac, not my 2021 M1 MacBook Pro). The message is as follows:

/Users/michellelong/.zprofile:2: no such file or directory: /opt/homebrew/bin/brew  
/Users/michellelong/.zprofile:9: no such file or directory: /opt/homebrew/bin/brew

I followed the same install steps on both computers, but I am only having this issue on the one.

### Install

I followed standard setup procedure:

```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
(echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> /Users/michellelong/.zprofile
(echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> /Users/michellelong/.zprofile
```

### Contents of .zprofile:

```
eval "$(/opt/homebrew/bin/brew shellenv)"

#Setting PATH for Python 3.11
#The original version is saved in .zprofile.pysave
PATH="/Library/Frameworks/Python.framework/Versions/3.11/bin:${PATH}"
export PATH

eval "$(/opt/homebrew/bin/brew shellenv)"
eval "$(/usr/local/bin/brew shellenv)"
```

### brew doctor

I ran the only check I know how to do: brew doctor AND brew --prefix Returns: Your system is ready to brew. AND /usr/local

### cd to op/homebrew

On MacBook shows a stable status On iMac no status appears, and although I can cd into the directories, they are empty.

### Uninstall and reinstall

I uninstalled and reinstalled on iMac, hoping that would clear things up, but I am in exactly the same situation. To note: the second install did not require the Next Steps that the first install did.

I know I am probably missing something simple, bit I am new to this and don't know what to do. Thanks in advance.

**Comment by SMillerDev** [comment, marked as answer]
So is your iMac an intel machine? Because those have Homebrew in `/use/local` not `/opt/Homebrew`

**Reply by michellehutton** [reply]
Yes, it is an intel machine. So, if the directory that it is saying doesn't exist doesn't need to exist, why would the terminal give me these messages every time I open it?

**Reply by SMillerDev** [reply]
Because you didn't read the messages at the end of the install correctly and just copied the instructions from your Arm machine  and now zprofile is trying to source `/opt/Homebrew/bin/brew shellenv` while it should be sourcing `/usr/local/bin/brew shellenv`.

**Reply by anwfeb** [reply]
**@SMillerDev** 
yes this works for me, thank you this information is very useful

**Reply by dima-devs** [reply]
@SMillerDev It works. Thanks for your help, saved my time.

**Reply by Nelbarretoc** [reply]
Just copy and paste:   `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"` and follow the instructions.

**Comment by hosaypeng** [comment]
i also got this error (my iMac is intel). 

what's the solution?

**Reply by michellehutton** [reply]
I went into my .zprofile and changed the wrong source (`/opt/Homebrew/bin/brew shellenv`) to the correct one (`/usr/local/bin/brew shellenv`). This fixed the issue for me.

**Reply by jay-f-k** [reply]
please how did you change it? am having same experience

**Reply by michellehutton** [reply]
Open your .zprofile file either through the terminal or GUI and look for the `/opt/Homebrew/bin/brew shellenv` line. Change that line to `/usr/local/bin/brew shellenv`. Just delete the old line and paste in the new one.

If you are having trouble finding the .zprofile file, here is how to find it on a Mac using the GUI:
1) Open your home folder in Finder. 
2) Click the Shift + CMD + . keys on your keyboard to show hidden files. 
3) Look for the .zprofile file and open it.

**Reply by Chasejlaw** [reply]
Thanks @michellehutton, this fixed my persistent issue as well!

**Reply by BhanuVikram** [reply]
Thank you, @michellehutton!

**Reply by HackBauer6611** [reply]
Thanks so much, @michellehutton, @SMillerDev! This helped me install homebrew right for my system the second time and remove my homebrew lines in my zprofile folder.

**Reply by Rodriguessz** [reply]
Thanks! Thanks! This solves my problem.

**Reply by ShailSri** [reply]
Thank you very much, @michellehutton.

**Comment by Quratul12** [comment]
I also had such error but thank god later i fixed my issue

**Comment by andrewpeters9** [comment]
In my case, I solved it by re-installing direnv:
```bash
brew install direnv
```
